src/main/java/edu/ucla/library/iiif/fester/handlers/EndpointConfigHandler.java

```diff
@@ -0,0 +1,198 @@
+
+import edu.ucla.library.iiif.fester.Constants;
+import edu.ucla.library.iiif.fester.Features;
+import edu.ucla.library.iiif.fester.HTTP;
+import edu.ucla.library.iiif.fester.MessageCodes;
+import edu.ucla.library.iiif.fester.Op;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.api.contract.openapi3.OpenAPI3RouterFactory;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.ext.web.handler.StaticHandler;
+
+/**
+ * An OpenAPI endpoint configuration handler.
+ */
+public class EndpointConfigHandler implements Handler<AsyncResult<OpenAPI3RouterFactory>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EndpointConfigHandler.class, Constants.MESSAGES);
+
+    private static final String FEATURE_FLAGS_FILE = "/etc/fester/fester-features.conf";
+
+    private static final String BATCH_UPLOAD_FORM = "/fester/upload/csv";
+
+    private final Promise<Router> myPromise;
+
+    private final JsonObject myConfig;
+
+    private final Vertx myVertx;
+
+    /**
+     * Creates a new endpoint configuration handler.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aConfig An application configuration
+     * @param aPromise A router configuration promise
+     */
+    public EndpointConfigHandler(final Vertx aVertx, final JsonObject aConfig, final Promise<Router> aPromise) {
+        myPromise = aPromise;
+        myConfig = aConfig;
+        myVertx = aVertx;
+    }
+
+    @Override
+    @SuppressWarnings("checkstyle:indentation") // Checkstyle can't handle the ifPresentOrElse syntax
+    public void handle(final AsyncResult<OpenAPI3RouterFactory> aConfiguration) {
+        final Optional<FeatureFlagChecker> featureFlagChecker = getFeatureFlagChecker();
+
+        if (aConfiguration.succeeded()) {
+            final OpenAPI3RouterFactory factory = aConfiguration.result();
+            final Promise<Boolean> featureConfigPromise = Promise.promise();
+
+            // We need to associate endpoint handlers with routes from our specification
+            factory.addHandlerByOperationId(Op.GET_STATUS, new GetStatusHandler());
+            factory.addHandlerByOperationId(Op.GET_MANIFEST, new GetManifestHandler(myVertx, myConfig));
+            factory.addHandlerByOperationId(Op.PUT_MANIFEST, new PutManifestHandler(myVertx, myConfig));
+            factory.addHandlerByOperationId(Op.DELETE_MANIFEST, new DeleteManifestHandler(myVertx, myConfig));
+            factory.addHandlerByOperationId(Op.GET_COLLECTION, new GetCollectionHandler(myVertx, myConfig));
+            factory.addHandlerByOperationId(Op.PUT_COLLECTION, new PutCollectionHandler(myVertx, myConfig));
+
+            // After the batch ingest feature is configured (or not), we complete the router configuration
+            featureConfigPromise.future().setHandler(handler -> {
```

Comment:
The only reason I'm raising this is because I think it hinders code readability -- it took me a little while to wrap my head around what `featureConfigPromise` was for. I think it would be clearer to instead fail `myPromise` straight from `configureBatchIngest`.

Reply:
Turns out I was mistaken, it looks like my proposed change wouldn't be an improvement. Thanks @ksclarke for walking me through